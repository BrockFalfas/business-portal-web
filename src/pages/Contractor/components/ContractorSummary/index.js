import React from 'react';
import PropTypes from 'prop-types';
import { formatUsd } from '~utils/number';

import './ContractorSummary.css';

const ContractorSummary = ({ summary }) => {
  return (
    <div className="ContractorSummary">
      <div className="ContractorSummary-labels">
        <div>Rank</div>
        <div>Num Jobs</div>
        <div>Prev</div>
        <div>Current</div>
        <div>YTD EARNINGS</div>
      </div>
      <div className="ContractorSummary-values">
        <div>{summary.rank}</div>
        <div>{summary.numOfJobs}</div>
        <div>{formatUsd(summary.prev)}</div>
        <div>{formatUsd(summary.current)}</div>
        <div>{formatUsd(summary.yearly)}</div>
      </div>
    </div>
  );
};

ContractorSummary.propTypes = {
  summary: PropTypes.shape({
    rank: PropTypes.number.isRequired,
    numOfJobs: PropTypes.number.isRequired,
    prev: PropTypes.number.isRequired,
    current: PropTypes.number.isRequired,
    yearly: PropTypes.number.isRequired,
  }),
};

export default ContractorSummary;
