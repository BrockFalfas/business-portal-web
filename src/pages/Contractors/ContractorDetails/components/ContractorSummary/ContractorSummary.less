@import "../../../../../styles/variables.less";

.ContractorSummary {
  margin-bottom: 2rem;

  &-labels > div,
  &-values > div {
    flex: 2;

    &:first-child {
      flex: 1;
    }
  }

  &-labels {
    display: flex;
    text-align: center;

    & > div {
      padding-top: 0.5rem;
      padding-bottom: 0.5rem;
      font-weight: bold;
    }
  }

  &-values {
    display: flex;
    background: @color-darker-blue;
    background: @gradient-blue;
    border-radius: 0.5rem;
    height: 3.5rem;
    line-height: 3.5rem;
    color: @color-white;
    text-align: center;
  }
}
