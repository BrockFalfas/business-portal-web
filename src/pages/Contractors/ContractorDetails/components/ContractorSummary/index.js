import React from 'react';
import { formatUsd } from '~utils/number';

import './ContractorSummary.css';

const ContractorSummary = ({ rank, nJobs, prev, current, ytd }) => {
  return (
    <div className="ContractorSummary">
      <div className="ContractorSummary-labels">
        <div>Rank</div>
        <div>Num Jobs</div>
        <div>Prev</div>
        <div>Selected Period</div>
        <div>YTD EARNINGS</div>
      </div>
      <div className="ContractorSummary-values">
        <div>{rank}</div>
        <div>{nJobs}</div>
        <div>{formatUsd(prev)}</div>
        <div>{formatUsd(current)}</div>
        <div>{formatUsd(ytd)}</div>
      </div>
    </div>
  );
};

export default ContractorSummary;
