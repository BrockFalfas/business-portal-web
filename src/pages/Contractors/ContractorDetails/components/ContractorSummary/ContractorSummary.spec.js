import React from 'react';
import { shallow } from 'enzyme';

import ContractorSummary from './index';

const initContractorSummary = overrides => {
  const mockProps = {
    rank: 1,
    nJobs: 10,
    prev: 1234.56,
    current: 987.65,
    ytd: 12345.67,
  };

  const wrapper = shallow(<ContractorSummary {...mockProps} {...overrides} />);
  return { mockProps, wrapper };
};

describe('component: <ContractorSummary/>', () => {
  describe('ContractorSummary', () => {
    it('should render without crashing', () => {
      const { wrapper } = initContractorSummary();

      expect(wrapper).toBeTruthy();
    });
  });

  it('render ContractorSummary', () => {
    const { wrapper } = initContractorSummary();

    expect(wrapper).toMatchSnapshot();
  });
});
