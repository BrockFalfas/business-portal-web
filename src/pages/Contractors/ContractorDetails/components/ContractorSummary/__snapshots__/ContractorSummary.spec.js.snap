// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: <ContractorSummary/> render ContractorSummary 1`] = `
<div
  className="ContractorSummary"
>
  <div
    className="ContractorSummary-labels"
  >
    <div>
      Rank
    </div>
    <div>
      Num Jobs
    </div>
    <div>
      Prev
    </div>
    <div>
      Selected Period
    </div>
    <div>
      YTD EARNINGS
    </div>
  </div>
  <div
    className="ContractorSummary-values"
  >
    <div>
      1
    </div>
    <div>
      10
    </div>
    <div>
      $1,234.56
    </div>
    <div>
      $987.65
    </div>
    <div>
      $12,345.67
    </div>
  </div>
</div>
`;
